examples/secrets-management-with-csi-secrets-driver-aws/README.md

```diff
@@ -0,0 +1,131 @@
+ - Creates a new sample VPC, 3 Private Subnets and 3 Public Subnets
+ - Creates Internet gateway for Public Subnets and NAT Gateway for Private Subnets
+ - Provisions AWS Secrets Manager and Config Provider for Secret Store CSI Driver
+ - Creates a namespace , IAM Roles for the service account, and the required CRDs required to retrieve the secrets for the application
+ - Deploys a sample pod to demonstrate mounting of secrets retrieved from AWS Secrets Manager as CSI Volume
+ 
+
+## How to Deploy
+### Prerequisites:
+
+Ensure that you have installed the following tools in your Mac or Windows Laptop before start working with this module and run Terraform Plan and Apply
+1. [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+2. [Kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+#### Secrets Management
+1. Create the secrets in [AWS Secrets Manager] (https://docs.aws.amazon.com/secretsmanager/latest/userguide/managing-secrets.html)
+2. Copy the ARN of the secrets and paste it under `secretsconfig.yaml` . `secretsconfig.yaml` represents `spec.parameters.objects` of [SecretProviderClass](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_csi_driver.html#integrating_csi_driver_SecretProviderClass) . 
+
+*Note* At this time, this example supports use cases which refers the secrets using 'A secret specified by full ARN'. If you refer the secret using name, the deployment will fail.
+
+### Deployment Steps
+#### Step1: Clone the repo using the command below
+
+```shell script
+git clone https://github.com/aws-ia/terraform-aws-eks-blueprints.git
+```
+
+#### Step2: Run Terraform INIT
+Initialize a working directory with configuration files
+
+```shell script
+cd examples/secrets-management-with-csi-secrets-driver-aws/
+terraform init
+```
+
+#### Step3: Run Terraform PLAN
+Verify the resources created by this execution
+
+```shell script
+export AWS_REGION=<ENTER YOUR REGION>   # Select your own region
+terraform plan
+```
+
+#### Step4: Finally, Terraform APPLY
+to create resources
+
+```shell script
+terraform apply
+```
+
+Enter `yes` to apply
+
+### Configure `kubectl` and test cluster
+EKS Cluster details can be extracted from terraform output or from AWS Console to get the name of cluster.
+This following command used to update the `kubeconfig` in your local machine where you run kubectl commands to interact with your EKS Cluster.
+
+#### Step5: Run `update-kubeconfig` command
+
+`~/.kube/config` file gets updated with cluster details and certificate from the below command
+
+    $ aws eks --region <enter-your-region> update-kubeconfig --name <cluster-name>
+
+#### Step6: View the `SecretProviderClass`
+
+    $ kubectl get SecretProviderClass -n <application>
+
+#### Step7: Exec in the Pod to view the secrets getting mounted successfully
+
+    $ kubectl exec -it -n nginx pod/nginx-secrets-pod-sample bash
```

Comment:
how is this pod deployed? and conversely, what secrets are we expecting to see if the [secretconfig](https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/471/files#diff-948ff2266135aa06316f7d25374d5c4ef3eda95c9fab5827fb2f305caa099a4bR3) does not contain any valid secret ARNs?

Reply:
Pod is deployed in the example itself as a sample nginx container. I removed the need of secretconfig now in the sample as i am creating the secret in the example itself.